src/mod.ts

```diff
@@ -29,13 +29,19 @@ export type EmojiFlavor<C extends Context = Context> = C & {
     replyWithEmoji: (
         string: TemplateStringsArray,
         ...emojis: EmojiName[]
-    ) => ReturnType<C['reply']>;
+    ) => ReturnType<C["reply"]>;
 };
 
 function withEmoji(string: TemplateStringsArray, ...emojis: EmojiName[]) {
     return string.reduce((acc, str, index) => {
+        // ! the number of elements in `string` is 1 more than in `emojis`,
+        // ! so we will look for `undefined` emoji, which does not make sense,
+        // ! so we exit the loop early.
+        if (index === string.length - 1) return acc + str;
         const emoji = getEmoji(emojis[index]);
-        return acc + str + (emoji ? emoji.emoji : "");
+        // ! `emoji` can be a string type if it's a custom value
+        // ! or an object if it's an emoji
+        return acc + str + (typeof emoji === "string" ? emoji : emoji.emoji);
```

Comment:
I'm no longer sure if reduce is the most readable choice here, but I guess it works

Reply:
In general, I would rewrite the plugin a little bit, because, for example, when I was dealing with it, it took me quite a long time to understand it

Reply:
Fair enough. That would be unrelated, so I guess we can merge this as-is, and then you can do this refactoring in a subsequent PQ.